### PR TITLE
Update `working with javascript` readme to support rails-ujs behaviour.

### PR DESCRIPTION
**Pull request by DmytroVasin**
### Summary
Small addition to the Docs about current Rails-ujs event handlers.

As was discussed in the next stories - it would be good to have such description:

https://github.com/rails/rails/issues/29046
https://github.com/rails/rails/issues/29137

**Please Note:**
Stoppable event in rails-ujs does not work, so I did not add documentation about them.

[Stoppable Events](http://guides.rubyonrails.org/working_with_javascript_in_rails.html#stoppable-events)

Thanks.

### CODE REVIEW

File: guides/source/working_with_javascript_in_rails.md [hunk 1]
```diff
@@ -376,6 +376,34 @@ browser to submit the form via normal means (i.e. non-AJAX submission) will be
 canceled and the form will not be submitted at all. This is useful for
 implementing your own AJAX file upload workaround.
 
+### Rails-ujs event handlers
+
+In Rails 5.1 was introduced rails-ujs which drop jQuery as a dependency.
```
Comment (vipulnsward) [inline guides/source/working_with_javascript_in_rails.md]: Rails 5.1 introduced rails-ujs and dropped jQuery as a dependency.

File: guides/source/working_with_javascript_in_rails.md [hunk 2]
```diff
@@ -376,6 +376,34 @@ browser to submit the form via normal means (i.e. non-AJAX submission) will be
...
+As a result the Unobtrusive JavaScript (UJS) driver has been rewritten to operate without jQuery.
+These introduction cause small changes into `custom events` fired during request:
```
Comment (vipulnsward) [inline guides/source/working_with_javascript_in_rails.md]: `introductions`

Comment (vipulnsward) [inline guides/source/working_with_javascript_in_rails.md]: cause small changes to `custom events` fired during the request:

File: guides/source/working_with_javascript_in_rails.md [hunk 3]
```diff
@@ -376,6 +376,34 @@ browser to submit the form via normal means (i.e. non-AJAX submission) will be
...
+
+NOTE: Signature of calls to UJS’s event handlers has changed.
```
Comment (vipulnsward) [inline guides/source/working_with_javascript_in_rails.md]: have changed

File: guides/source/working_with_javascript_in_rails.md [hunk 4]
```diff
@@ -376,6 +376,34 @@ browser to submit the form via normal means (i.e. non-AJAX submission) will be
...
+Unlike the version with jqeury-ujs, all custom events return only one parameter: `event`.
```
Comment (vipulnsward) [inline guides/source/working_with_javascript_in_rails.md]: jquery

File: guides/source/working_with_javascript_in_rails.md [hunk 5]
```diff
@@ -376,6 +376,34 @@ browser to submit the form via normal means (i.e. non-AJAX submission) will be
...
+In this parameter there is an additional attribute `details` which contains an array of extra parameters.
```
Comment (vipulnsward) [inline guides/source/working_with_javascript_in_rails.md]: comma after parameter

File: guides/source/working_with_javascript_in_rails.md [hunk 6]
```diff
@@ -376,6 +376,34 @@ browser to submit the form via normal means (i.e. non-AJAX submission) will be
...
+|---------------------|---------------------------------|-------------------------------------------------------------|
+| `ajax:before`       |                                 | Before the whole ajax business.                             |
+| `ajax:beforeSend`   | [xhr, options]                  | Before the request is sent.                                 |
+| `ajax:send`         | [xhr]                           | When the request is sent.                                   |
+| `ajax:stopped`      |                                 | When the request is stopped.                                |
+| `ajax:success`      | [response, status, xhr]         | After completion, if the response was a success.            |
+| `ajax:error`        | [response, status, xhr]         | After completion, if the response was an error.             |
+| `ajax:complete`     | [xhr, status]                   | After the request has been completed, no matter the outcome.|
+
+Example of usage:
```
Comment (vipulnsward) [inline guides/source/working_with_javascript_in_rails.md]: Example usage:

File: guides/source/working_with_javascript_in_rails.md [hunk 7]
```diff
@@ -376,6 +376,34 @@ browser to submit the form via normal means (i.e. non-AJAX submission) will be
...
+
+```html
+document.body.addEventListener('ajax:success', function(event) {
+  ref = event.detail, data = ref[0], status = ref[1], xhr = ref[2];
```
Comment (vipulnsward) [inline guides/source/working_with_javascript_in_rails.md]:
```javascript
var detail = event.detail; 
var data = detail[0], status = detail[1],  xhr = detail[2];
```

**Review by vipulnsward — CHANGES_REQUESTED** [review]
Thanks for the changes!

I have added some grammar edits to be made.

**Comment by DmytroVasin** [comment]
@vipulnsward Done.

Thanks for corrections

**Review by vipulnsward — APPROVED** [review]

**Comment by vipulnsward** [comment]
Thanks, we can polish it over. 

Please make sure to add [ci skip] in future for doc only changes.

**Pull request merged**